patches/server/1055-Fix-creation-of-invalid-block-entity-during-world-ge.patch

```diff
@@ -0,0 +1,40 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Pierpaolo Coletta <p.coletta@glyart.com>
+Date: Sat, 30 Mar 2024 21:06:10 +0100
+Subject: [PATCH] Fix creation of invalid block entity during world generation
+
+
+diff --git a/src/main/java/net/minecraft/server/level/WorldGenRegion.java b/src/main/java/net/minecraft/server/level/WorldGenRegion.java
+index 5ece375eaf6bcc61864997a389bb5e24625e4505..9c3f8f79c2b3389a118dce9a1558edda52446833 100644
+--- a/src/main/java/net/minecraft/server/level/WorldGenRegion.java
++++ b/src/main/java/net/minecraft/server/level/WorldGenRegion.java
+@@ -352,6 +352,7 @@ public class WorldGenRegion implements WorldGenLevel {
+                         ichunkaccess.removeBlockEntity(pos);
+                     }
+                 } else {
++                    ichunkaccess.removeBlockEntity(pos); // Paper - Clear the block entity before setting up a DUMMY block entity
```

Comment:
but, we're just about to add a new entry, how would this change anything?

Reply:
The entry is added only to the pendingBlockEntities map, by default if there is something in the real blockEntities map it stays there (at least for ProtoChunks)

The issue is that if we use the AsyncStructureGenerateEvent with a block transformer it's going to access the BlockEntity and so WorldGenRegion creates it, but if that block is later changed the old block entity is still there because it's not removed from the blockEntities map

Check WorldGenRegion#getBlockEntity